src/instructlab/model/linux_test.py

```diff
@@ -0,0 +1,73 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+from typing import Optional
+import logging
+import os
+
+# Third Party
+from datasets import load_dataset
+from openai import OpenAI, Stream
+from tqdm import tqdm
+import click
+
+# Local
+from .. import configuration as cfg
+from ..utils import get_sysprompt
+
+
+def linux_test(
+    ctx: click.Context,
+    api_base,
+    test_file: Path,
+    api_key: Optional[str] = None,
+    models: list = [cfg.DEFAULT_MODEL],
+    create_params: dict = {},
+):
+    def response(user, create_params):
+        resp = client.chat.completions.create(
+            **create_params,
+            messages=[
+                {"role": "system", "content": get_sysprompt()},
+                {"role": "user", "content": user},
+            ],
+        )
+        if isinstance(resp, Stream):
+            # https://platform.openai.com/docs/api-reference/chat/streaming
+            response = ""
+            for r in resp:
+                if c := r.choices[0].delta.content:
+                    print(c, end="", flush=True)
+                    response += c
+            print()
+            return response
+        else:
+            # https://platform.openai.com/docs/api-reference/chat/object
+            c = resp.choices[0].message.content
+            print(c)
+            return c
+
+    ctx.obj.logger.debug("")
+
+    client = OpenAI(
+        base_url=api_base,
+        api_key=api_key,
+    )
+
+    # https://platform.openai.com/docs/api-reference/chat/create
+    create_params["temperature"] = 0  # for more reproducible results
+    create_params["stream"] = True  # optional
```

Comment:
I think no need to use stream for test. Streaming is meaningful for chat but no need for test and generate.